dvc/main.py

```diff
@@ -81,6 +84,13 @@ def main(argv=None):
     if ret != 0:
         logger.info(FOOTER)
 
-    Analytics().send_cmd(cmd, args, ret)
+    if analytics.is_enabled():
+        report = analytics.collect(args, ret)
+
+        # Use a temporary file to pass the report between the current and
+        # detached process.
+        with tempfile.NamedTemporaryFile(delete=False, mode="w") as fobj:
+            json.dump(report, fobj)
+            daemon(["analytics", fobj.name])
```

Comment:
This seems to belong to the analytics.py and not the main.py.

Reply:
@efiop , I'd argue that)
It is using the analytics module to `collect` the data and then `send` it asynchronously. But the _sending_ part is more like a _hack_.

I guess it wouldn't feel that inappropriate to have an interface like that if we had a more pythonic mechanism of doing something "detached" or "daemonized", for example:

```python
report = analytics.colect(args, ret)
daemonize(analytics.send, report)
```

Reply:
Why not just
```
analytics.collect_and_send(args, ret)
```
? All of this analytics-related internals don't really belong to `main.py`, no?

Reply:
Isn't sending part also `analytics` responsibility? I would say that `main`'s purpose is just to invoke a proper method, and analytics should take care of its responsibility, that is gathering stats, sending it and deleting the remnants. 

Having said so, I think that leaving `analytics.is_enabled` in main is a good idea, just to make it clear to more inquisitive users that we are checking that analytics is enabled.

Reply:
@efiop, @pared, I changed the analytics daemon to receive the _command class_ and _return code_, now, it the report would be collected and send on the detached process.

I think it is more clean, I can't see any problems with this approach, maybe there was a reason for having a file instead of passing the arguments.